src/api/telemetry/latestValueSubscription.js

```diff
@@ -0,0 +1,177 @@
+     * @param {Function} callback the callback to invoke with new data, as
+     *        it becomes available
+     * @returns {Function} a function which may be called to terminate
+     *          the subscription
+     */
+    function latestValueSubscription(
+        domainObject,
+        callback,
+        telemetryAPI,
+        openmct
+    ) {
+        var latestDatum;
+        var pendingRealtimeDatum;
+        var currentRequest = 0;
+        var metadata = telemetryAPI.getMetadata(domainObject);
+        var formatters = telemetryAPI.getFormatMap(metadata);
+        var timeFormatter;
+        var active = true;
+        var restrictToBounds = false;
+
+        function isLater(a, b) {
+            return !timeFormatter || timeFormatter.parse(a) > timeFormatter.parse(b);
+        }
+
+        function applyBoundsFilter(datum) {
+            if (withinBounds(datum)) {
+                callback(datum);
+            }
+        }
+
+        function withinBounds(datum) {
+            if (!restrictToBounds || !timeFormatter) {
+                return true;
+            }
+            var timestamp = timeFormatter.parse(datum);
+            var bounds = openmct.time.bounds();
+            if (timestamp >= bounds.start && timestamp <= bounds.end) {
+                return true;
+            }
+        }
+
+        function updateClock(clock) {
+            // We restrict to bounds if there is no clock (aka fixed mode) to
+            // prevent users from seeing data they don't expect.
+            restrictToBounds = !clock;
+        }
+
+        function callbackIfLatest(datum) {
+            if (!active) {
+                return; // prevent LAD notify after unsubscribe.
+            }
+            // If we don't have latest data, store datum for later processing.
+            if (typeof latestDatum === 'undefined') {
+                if (typeof pendingRealtimeDatum === 'undefined' || (
+                        isLater(datum, pendingRealtimeDatum) &&
+                        withinBounds(datum)
+                    )) {
+
+                    pendingRealtimeDatum = datum;
+                }
+                return;
+            }
+            // If there is no latest data, or datum is latest, then notify
+            // subscriber.
+            if (latestDatum === false || isLater(datum, latestDatum)) {
+                latestDatum = datum;
+                applyBoundsFilter(datum);
+            }
+        }
+
+        function updateTimeSystem(timeSystem) {
+            // Reset subscription state, request new latest data and wait for
+            // response before filtering lad data.
+            latestDatum = undefined;
+            pendingRealtimeDatum = undefined;
+            timeFormatter = formatters[timeSystem.key];
+
+            currentRequest++;
+            var thisRequest = currentRequest;
+            telemetryAPI.request(domainObject, {strategy: 'latest', size: 1})
+                .then(function (results) {
```

Comment:
Maybe creating a cb func for each of these promises locally would make it more readable?